### PR TITLE
etcdctl: use health endpoint to greatly simplify health checking

### PR DESCRIPTION
**Pull request by xiang90**
Make cluster health checking more reliable and simple

### CODE REVIEW

File: etcdctl/command/cluster_health.go [hunk 1]
```diff
@@ -42,124 +40,61 @@ func handleClusterHealth(c *cli.Context) {
...
-			fmt.Print(p)
+			fmt.Println("cluster is unhealthy")
 		}
 
 		if !forever {
-			return
+			break
 		}
-
+		fmt.Printf("\nnext check after 10 second...\n\n")
```
Comment (yichengq) [inline etcdctl/command/cluster_health.go]: could we print out wall time for forever case? it helps users to check health history

File: etcdctl/command/cluster_health.go [hunk 2]
```diff
@@ -42,124 +40,61 @@ func handleClusterHealth(c *cli.Context) {
...
+				resp.Body.Close()
+				if err != nil {
+					fmt.Printf("failed to check the health of member %s on %s: %v\n", m.ID, url, err)
+					continue
+				}
+
+				checked = true
+				if result.Health == "true" {
+					health = true
+					checked = true
```
Comment (yichengq) [inline etcdctl/command/cluster_health.go]: checked has been set to true 3 lines before.

File: etcdctl/command/cluster_health.go (hunk 1 above)
Comment (xiang90) [inline etcdctl/command/cluster_health.go]: I think we have discussed about this issue before. We will keep etcdctl as it is toady. We might use a better logging pkg in the future.

File: etcdctl/command/cluster_health.go (hunk 2 above)
Comment (xiang90) [inline etcdctl/command/cluster_health.go]: sure

File: etcdctl/command/cluster_health.go (hunk 1 above)
Comment (yichengq) [inline etcdctl/command/cluster_health.go]: I forgot it. ok.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**